### PR TITLE
doc/user: Polish release notes for v0.22.0

### PR DESCRIPTION
**Pull request by morsapaes**
### Motivation

Weekly release notes PR.

### Tips for reviewer

* Skipped #10867, since it's been added to improve driver compatibility and the comment below makes me think it equates to something behind `--experimental`:

> I've intentionally left the `TimeZone` support relatively basic, as it will probably have to be changed substantially whenever we decide to add real support to timezones.

* Was eager to add #10764, but AFAIU it's been reverted in #10979, so the release note loses effect.

<hr>

~Side note: noticed that #10734 has been merged but the corresponding documentation PR (#10848) is still pending.~ (This has been merged!)

### CODE REVIEW

File: doc/user/content/release-notes.md [hunk 1]
```diff
@@ -19,6 +19,69 @@ the PR description instead. They will be migrated here during the release
...
+
+- Change the range of the `oid` type from [-231, 231 - 1] to [0, 232 - 1] to match PostgreSQL.
+
+- Fix a panic when materializing a source that has been renamed since it was last materialized {{% gh 10904 %}}.
+
+- Change the claimed PostgreSQL version returned by the `version()` function to 9.5 to match the values of the `server_version` and `server_version_num session` parameters.
+
+- Fix data loss bug of in [Postgres sources] introduced in version v0.20.0 {{% gh 10981 %}}.
+
+- More precisely infer the output type of numeric division and modulo operations as `NOT NULL` when the input arguments are `NOT NULL`.
```
Comment (benesch) [inline doc/user/content/release-notes.md:78]: This is a little in the weeds. I'll leave to you to though, @morsapaes, to decide whether to toss.

File: doc/user/content/release-notes.md [hunk 2]
```diff
@@ -19,6 +19,69 @@ the PR description instead. They will be migrated here during the release
...
+
+- Allow specifying [command line flags](/cli/) multiple times. The last
```
Comment (benesch) [inline doc/user/content/release-notes.md:80]:
```suggestion
- Allow specifying the same [command line flag](/cli/) multiple times. The last
```

(Sorry, I wrote this one too. 🙈)

File: doc/user/content/release-notes.md [hunk 3]
```diff
@@ -19,6 +19,69 @@ the PR description instead. They will be migrated here during the release
...
+specification takes precedence. This matches the behavior of many standard
+Unix tools and is particularly useful for folks using `materialized` via
+Docker.
```
Comment (benesch) [inline doc/user/content/release-notes.md:83]: Maybe a smidge more context?

```suggestion
Docker as it allows overwriting the default `--log-file` option.
```

File: doc/user/content/release-notes.md [hunk 4]
```diff
@@ -19,6 +19,69 @@ the PR description instead. They will be migrated here during the release
...
+
+  Previously `current_timestamp`, `mz_logical_timestamp`, and `mz_uptime` were
+  incorrectly disallowed in unmaterialized views, while the remaining nonpure
+  functions were incorrectly allowed in materialized views.
+
+- **Breaking change.** Store days separately in [`interval`].
+Unlike the previous version, Hours are never converted to days. This
+means that: an Interval of 24 hours will not be equal to an Interval
+of 1 day, you cannot subtract hours from days, and when ordering
+Intervals `d days > h hours` for all `d,h` {{% gh 10708 %}}.
```
Comment (benesch) [inline doc/user/content/release-notes.md:54]:
```suggestion
Unlike in previous versions, hours are not automatically converted to days. This
means that: an interval of 24 hours will not be equal to an interval
of 1 day, you cannot subtract hours from days, and when ordering
intervals `d days > h hours` for all `d` and `h` {{% gh 10708 %}}.

To force a conversion from hours to days, use the new [`justify_hours`](/sql/functions/justify-hours) function.
```

File: doc/user/content/release-notes.md [hunk 5]
```diff
@@ -19,6 +19,69 @@ the PR description instead. They will be migrated here during the release
...
+units as plural (e.g., `-1 days` will be printed instead of `-1 day`). This
+matches the behavior of PostgreSQL.
+
+- **Breaking change.** Round microsecond field of
+[`interval`] to 6 places before applying the
+given precision. For example `INTERVAL '1.2345649' SECOND(5)` will be
+rounded to `00:00:01.23457`, not `00:00:01.23456`. This matches the
+behavior of PostgreSQL.
+
+- Add the following aliases to Interval parsing: `yr`, `yrs`, `hr`, `hrs`, `min`, `mins`, `sec`, `secs`.
```
Comment (benesch) [inline doc/user/content/release-notes.md:66]:
```suggestion
- Add several new time units to [`interval`](/sql/types/interval) parsing: `yr`, `yrs`, `hr`, `hrs`, `min`, `mins`, `sec`, and `secs`.
```

Naturally, we forgot to update the docs, too: https://preview.materialize.com/10991/sql/types/interval/. The time units are listed in the time-unit.svg diagram.

File: doc/user/content/release-notes.md [hunk 6]
```diff
@@ -19,6 +19,69 @@ the PR description instead. They will be migrated here during the release
...
+
+- Add the [`justify_days`](/sql/functions/justify-days), [`justify_hours`](/sql/functions/justify-hours), and [`justify_interval`](/sql/functions/justify-interval) functions.
+
+- Change the range of the `oid` type from [-231, 231 - 1] to [0, 232 - 1] to match PostgreSQL.
```
Comment (benesch) [inline doc/user/content/release-notes.md:70]:
```suggestion
- Change the range of the `oid` type from [-2<sup>31</sup>, 2<sup>31</sup> - 1] to [0, 2<sup>32</sup> - 1] to match PostgreSQL.
```

File: doc/user/content/release-notes.md [hunk 7]
```diff
@@ -19,6 +19,69 @@ the PR description instead. They will be migrated here during the release
...
+
+- Fix a panic when materializing a source that has been renamed since it was last materialized {{% gh 10904 %}}.
```
Comment (benesch) [inline doc/user/content/release-notes.md:72]: @morsapaes would this be clearer? Or is it too much?

```suggestion
- Fix a panic that could occur if you performed the following sequence of operations {{% gh 10904 %}}:
  - `CREATE SOURCE src ...`
  - `CREATE INDEX src_idx ON src ...`
  - `ALTER src RENAME TO new_src`
  - `DROP INDEX src_idx`
  - `CREATE INDEX new_src_idx ON new_src ...` 
```

**Review by benesch — COMMENTED** [review]
Thanks so very much, @morsapaes! I'm sorry you're bearing the brunt of the new release notes process. Feel free to kick the pain back my way, especially while we don't have any automation set up yet.

**Comment by benesch** [comment]
Oh, two other thoughts. The release note for #10766 got lost because of the transition to the new process:

```
- Support bare `FORMAT` with `INCLUDE KEY AS` when using the confluent schema
   registry {{% gh 10766 %}}.
```

Needs some wordsmithing though. I'm not totally sure what it means.

Also looks like #10734 should get a release note. Chances #10848 before v0.22 ships?

File: doc/user/content/release-notes.md (hunk 6 above)
Comment (morsapaes) [reply doc/user/content/release-notes.md:70]: Duh.

File: doc/user/content/release-notes.md (hunk 1 above)
Comment (morsapaes) [reply doc/user/content/release-notes.md:78]: Yeah, I was on the fence about this one but will drop it.

**Comment by morsapaes** [comment]
Thanks for the review, @benesch! I've blended your suggestions in and added a note for #10734:

> Add support for [named composite types](https://www.postgresql.org/docs/current/rowtypes.html). Unimplemented features are listed in the original issue {{% gh 10734 %}}.

I'll follow-up on the docs PR with @pH14, but if he's still out sick I'll just push my suggestions in and merge it so it goes out with the release.

Regarding #10766, I think I'll need to summon @guswynn. My understanding is that #10730 (v0.21.0) fixed an actual issue with `INCLUDE KEY AS` not working with `ENVELOPE UPSERT`, but this change is more of a...nicety (?) that allows you to write:

```sql
CREATE MATERIALIZED SOURCE bareformatconfluent
  FROM KAFKA BROKER '...' TOPIC '...'
  FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY '...'
  INCLUDE KEY AS named
  ENVELOPE UPSERT;
```

instead of:

```sql
CREATE MATERIALIZED SOURCE kvformatconfluent
  FROM KAFKA BROKER '...' TOPIC '...'
  KEY FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY '...'
  VALUE FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY '...'
  INCLUDE KEY AS named
  ENVELOPE UPSERT;
```

when using the CSR. I'm a bit confused, since AFAIR the convention for `INCLUDE KEY` is to explicitly specify `KEY FORMAT...VALUE FORMAT` (it's also how it's [documented](https://materialize.com/docs/sql/create-source/kafka/#key)). So either the documentation needs to be patched (and folks need a full-on decision tree to grok _all_ possible combinations of syntax), or I'm...not sure this grants a release note since it seems to contradict the "standard" for this feature. 

Maybe I just need to step away for a minute and get some fresh air to untangle my brain. 🚶‍♀️

File: doc/user/content/release-notes.md (hunk 5 above)
Comment (morsapaes) [reply doc/user/content/release-notes.md]: On my list.

**Comment by morsapaes** [comment]
Thanks for the run through, @guswynn! It's way clearer now; pushed in a release note! I'll adjust the Kafka docs as well and add an example.

**Comment by morsapaes** [comment]
@benesch, all tweaks pushed in, if you want to give it another look.

File: doc/user/content/release-notes.md [hunk 8]
```diff
@@ -19,6 +19,90 @@ the PR description instead. They will be migrated here during the release
...
+  incorrectly disallowed in unmaterialized views, while the remaining nonpure
+  functions were incorrectly allowed in materialized views.
+
+- **Breaking change.** Store days separately in [`interval`].
+Unlike in previous versions, hours are not automatically converted to days. This
+means that: an interval of 24 hours will not be equal to an interval
+of 1 day, you cannot subtract hours from days, and when ordering
+intervals `d days > h hours` for all `d` and `h` {{% gh 10708 %}}.
+
+To force a conversion from hours to days, use the new [`justify_hours`](/sql/functions/justify-hours) function.
```
Comment (benesch) [inline doc/user/content/release-notes.md:56]: Nit: this needs a smidge of indentation to get lined up in the bullet.

**Review by benesch — APPROVED** [review]
Thank you so much, @morsapaes!

**Pull request merged**